### PR TITLE
add small utility for reading json file into objects

### PR DESCRIPTION
**Pull request by Qi77Qi**
Recently, there were two places where I see people need to read a json file into an object A. So adding a small utility function for doing that

**PR checklist**
- [ ] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [ ] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ To depend on the `MockGoogle*` classes, additionally depend on:
 
 Contains utility functions for talking to Google APIs via com.google.cloud client library (more recent) via gRPC. 
 
-Latest SBT dependency: `"org.broadinstitute.dsde.workbench" %% "workbench-google2" % "0.3-e7d949f"`
+Latest SBT dependency: `"org.broadinstitute.dsde.workbench" %% "workbench-google2" % "0.2-TRAVIS-REPLACE-ME"`
```
Comment (jdcanas) [inline README.md]: previous value got clobbered

File: util/src/main/scala/org/broadinstitute/dsde/workbench/util/package.scala [hunk 2]
```diff
@@ -32,4 +40,17 @@ package object util {
...
   def readFile[F[_]](path: String)(implicit sf: Sync[F]): Resource[F, FileInputStream] =
     Resource.make(sf.delay(new FileInputStream(path)))(f => sf.delay(f.close()))
+
+  /*
+   * Example:
+   * scala> org.broadinstitute.dsde.workbench.util.readJsonFileToA[IO, List[String]](java.nio.file.Paths.get("/tmp/list"), None).compile.lastOrError.unsafeRunSync
+   * res0: List[String] = List(this is great)
+   *
+   */
+  def readJsonFileToA[F[_]: Sync: ContextShift: Concurrent, A: Decoder](path: Path, blockingExecutionContext: Option[ExecutionContext] = None): Stream[F, A] = {
```
Comment (jdcanas) [inline util/src/main/scala/org/broadinstitute/dsde/workbench/util/package.scala:50]: what is the behavior if the path passed isn't a valid path? Will the stream be 'empty' and it fail silently, or will it return some sort of throwable wrapped in the stream?

**Review by jdcanas — APPROVED** [review]
Looks good except for the hash clobber. I'm not too sure if the method added needs more error handling, its not clear to me the behavior if the path passed is invalid (and, more importantly, if the caller will receive a helpful message)

File: util/src/main/scala/org/broadinstitute/dsde/workbench/util/package.scala (hunk 2 above)
Comment (Qi77Qi) [reply util/src/main/scala/org/broadinstitute/dsde/workbench/util/package.scala:50]: the `F` will fail with appropriate error like
```
scala> org.broadinstitute.dsde.workbench.util.readJsonFileToA[IO, List[String]](java.nio.file.Paths.get("/tmp/list"), None).compile.lastOrError.unsafeRunSync
java.nio.file.NoSuchFileException: /tmp/list
  at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
  at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
  at sun.nio.fs.UnixException.rethrowAsIOException(
```

**Pull request merged**